### PR TITLE
feat: join a call when there is already an active call - PART 1 (AR-1664)

### PR DESCRIPTION
**Pull request by ohassine**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/AR-1664" title="AR-1664" target="_blank">StoryAR-1664</a>  User tries to make a call, when in a call already, calling UI appears
  </td></table>
  <br />
 

----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [x] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [x] contains a reference JIRA issue number like `SQPIT-764`
  - [x] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [x] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

### Description

This PR aims to handle the case of accepting an incoming call when we are already in an active call

The flow is like:
If the user tries to accept the  incoming call, then we display a dialog to inform him if he wants to end the current one and join the newer one
If he confirms then we hang up the current call and accept the new one

Needs releases with:

- [ ] GitHub link to other pull request

#### How to Test

When you are on call, try to accept another incoming call

----
#### PR Post Submission Checklist for internal contributors (Optional)

 - [ ] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

 - [ ] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

**Comment by AndroidBob** [comment]
Build [1](https://10.10.124.134/job/wire-android-reloaded/job/PR-750/1/) **failed**.

**Comment by AndroidBob** [comment]
Build [2](https://10.10.124.134/job/wire-android-reloaded/job/PR-750/2/) **failed**.

**Comment by AndroidBob** [comment]
Build [7](https://10.10.124.134/job/wire-android-reloaded/job/PR-750/7/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv0.0.1(3866279)(3866279)-dev-release.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/dev/release/com.wire.android-vv0.0.1(3866279)(3866279)-dev-release.apk)

**Comment by AndroidBob** [comment]
Build [9](https://10.10.124.134/job/wire-android-reloaded/job/PR-750/9/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv0.0.1(3866693)(3866693)-dev-release.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/dev/release/com.wire.android-vv0.0.1(3866693)(3866693)-dev-release.apk)

File: app/src/main/kotlin/com/wire/android/ui/home/conversations/ConversationViewModel.kt [hunk 1]
```diff
@@ -77,6 +78,7 @@ class ConversationViewModel @Inject constructor(
     private val getMessageForConversation: GetMessagesForConversationUseCase,
     private val isFileSharingEnabled: IsFileSharingEnabledUseCase,
     private val observeOngoingCalls: ObserveOngoingCallsUseCase,
+    private val observeEstablishedCallsUseCase: ObserveEstablishedCallsUseCase,
```
Comment (yamilmedina) [inline app/src/main/kotlin/com/wire/android/ui/home/conversations/ConversationViewModel.kt:81]:
```suggestion
    private val observeEstablishedCalls ObserveEstablishedCallsUseCase,
```

**Review by yamilmedina — COMMENTED** [review]
Added some comments about naming, in general great work! 👏

**Comment by AndroidBob** [comment]
Build [10](https://10.10.124.134/job/wire-android-reloaded/job/PR-750/10/) **failed**.

**Review by yamilmedina — APPROVED** [review]
lgtm! 🦅 👏

**Comment by AndroidBob** [comment]
Build [11](https://10.10.124.134/job/wire-android-reloaded/job/PR-750/11/) **failed**.

**Comment by AndroidBob** [comment]
Build [12](https://10.10.124.134/job/wire-android-reloaded/job/PR-750/12/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv0.0.1(3873354)(3873354)-dev-release.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/dev/release/com.wire.android-vv0.0.1(3873354)(3873354)-dev-release.apk)

**Pull request merged**